README.md

```diff
@@ -401,6 +405,98 @@ If the modification of the Application resource is completed successfully, the `
+
+If an Application Resource Mapping defines `containers`, it **MUST NOT** define `envs` and `volumeMounts`.  If an Application resources does not define `containers`, it **MUST** define `envs` and `volumeMounts`.
+
+[carm-crd]: service.binding_clusterapplicationresourcemappings.yaml
+[container]: https://kubernetes.io/docs/reference/kubernetes-api/workloads-resources/container/
+[crd-syntax]: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition
+[envvar]: https://kubernetes.io/docs/reference/kubernetes-api/workloads-resources/container/#environment-variables
+[jsonpath]: https://kubernetes.io/docs/reference/kubectl/jsonpath/
+[volume]: https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/volume
+[volumemount]: https://kubernetes.io/docs/reference/kubernetes-api/workloads-resources/container/#volumes
+
+### Resource Type Schema
+
+```yaml
+apiVersion: service.binding/v1alpha2
+kind: ClusterApplicationResourceMapping
+metadata:
+  name:                 # string
+  generation:           # int64, defined by the Kubernetes control plane
+  ...
+spec:
+  versions:             # []Version
+  - version:            # string
+    containers:         # []string, optional
+    envs:               # []string, optional
+    volumeMounts:       # []string, optional
+    volumes:            # string
+```
+
+### Container-based Example Resource
+
+```yaml
+apiVersion: service.binding/v1alpha2
+kind: ClusterApplicationResourceMapping
+metadata:
+ name:  cronjobs.batch
+spec:
+  versions:
+  - version: v1beta1
+    containers:
+    - .spec.jobTemplate.spec.template.spec.containers
+    - .spec.jobTemplate.spec.template.spec.initContainers
+    volumes: .spec.jobTemplate.spec.template.spec.volumes
+  - version: v2alpha1
+    containers:
+    - .spec.jobTemplate.spec.template.spec.containers
+    - .spec.jobTemplate.spec.template.spec.initContainers
+    volumes: .spec.jobTemplate.spec.template.spec.volumes
+```
+
+### Element-based Example Resource
+
+```yaml
+apiVersion: service.binding/v1alpha2
+kind: ClusterApplicationResourceMapping
+metadata:
+ name:  cronjobs.batch
+spec:
+  versions:
+  - version: v1beta1
+    env:
+    - .spec.jobTemplate.spec.template.spec.containers[*].env
+    - .spec.jobTemplate.spec.template.spec.initContainers[*].env
+    volumeMounts:
+    - .spec.jobTemplate.spec.template.spec.containers[*].volumeMounts
+    - .spec.jobTemplate.spec.template.spec.initContainers[*].volumeMounts
+    volumes: .spec.jobTemplate.spec.template.spec.volumes
+  - version: v2alpha1
+    env:
+    - .spec.jobTemplate.spec.template.spec.containers[*].env
+    - .spec.jobTemplate.spec.template.spec.initContainers[*].env
+    volumeMounts:
+    - .spec.jobTemplate.spec.template.spec.containers[*].volumeMounts
+    - .spec.jobTemplate.spec.template.spec.initContainers[*].volumeMounts
+    volumes: .spec.jobTemplate.spec.template.spec.volumes
+```
+
```

Comment:
Maybe also include a example for the default, behavior:

This will need to be apply manually because of the nested code block

```suggestion
### Default, PodSpecable Mapping

```yaml
apiVersion: service.binding/v1alpha2
kind: ClusterApplicationResourceMapping
metadata:
  name: "*" # not an actual resource
spec:
  versions:
  - version: "*"
    containers:
    - .spec.template.spec.containers
    - .spec.template.spec.initContainers
    volumes: .spec.template.spec.volumes
```
```